app/workers/metrics/record_db_and_es_record_counts_worker.rb

```diff
@@ -0,0 +1,64 @@
+# This worker checks the number of records for each Elasticsearch index and
+# compares it to the number of records in our database.
+#
+# For indexes that don't match a single model (no model or multiple models) we
+# need to implement a custom db_count method on the Search class to do the
+# counts for us
+module Metrics
+  class RecordDbAndEsRecordCountsWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :low_priority, retry: 5
+
+    # Once the Elasticsearch migration is complete we can update this value to
+    # Search::Cluster::SEARCH_CLASSES which will include all indexes.
+    SEARCH_CLASSES = [
+      Search::ChatChannelMembership,
+      Search::ClassifiedListing,
+      Search::Tag,
+    ].freeze
+
+    # Adjustable margin of error - this is how far off the index count can be
+    # from the database count before we raise an error
+    def perform(margin_of_error = 0)
```

Comment:
There's a chance we'll get some false errors running this. For example, if we run a query to count the records in the DB and then query ES for records, it's possible another record was created between those queries, making the counts not match. I added this `margin_of_error` argument to make it easy to adjust it in case our notifications get too noisy (change this to 3 or 5 for example).

It's certainly possible we want to extend this further by having `margin_of_error` be index specific. That's okay because we'd likely add that to the search class itself and we wouldn't have to update the arguments of this worker.